raami/queries.js

```diff
@@ -0,0 +1,269 @@
+                   description, transport, legal, auction, print, digital, agent, contact, waitlist, postage
+                 )
+          VALUES (
+                   $(person_id), $(name), $(continent), $(url), $(filename),
+                   $(filedata), $(description), $(transport), $(legal),
+                   ($(auction)), ($(print)), $(digital), $(agent), $(contact), $(waitlist), ($(postage))
+                 )
+       RETURNING id`, req.body
+  )
+    .then(function (data) {
+      console.log(data.id)
+      res.status(200)
+        .json({
+          status: 'success',
+          inserted: data.id
+        });
+    })
+    .catch(function (err) {
+      return next(err);
+    });
+}
+
+
+function updateArtist(req, res, next) {
+  var _id = (req.params.id);
+  req.app.locals.db.none(`
+    UPDATE Artist
+       SET continent=$1, url=$2, filename=$3, filedata=$4, name=$5,
+           description=$6, transport=$7, legal=$8, auction=$9, print=$10,
+           digital=$11, agent=$12, contact=$13, waitlist=$14, postage=$15
+     WHERE id=$16`, [
+      req.body.continent,
+      req.body.url,
+      req.body.filename,
+      req.body.filedata,
+      req.body.name,
+      req.body.description,
+      req.body.transport,
+      req.body.legal,
+      (req.body.auction),
+      (req.body.print),
+      req.body.digital,
+      req.body.agent,
+      req.body.contact,
+      req.body.waitlist,
+      (req.body.postage),
+      _id
+    ]
+  )
+    .then(function () {
+      res.status(200)
+        .json({
+          status: 'success',
+          updated: req.body
+        });
+    })
+    .catch(function (err) {
+      return next(err);
+    });
+}
+
+// function removeArtists(req, res, next) {
+//   var _id = parseInt(req.params.id);
+//   db.result('DELETE FROM Artist WHERE id = $1', _id)
+//     .then(function (result) {
+//       /* jshint ignore:start */
+//       res.status(200)
+//         .json({
+//           status: 'success',
+//           message: `Removed ${result.rowCount} puppy`
+//         });
+//       /* jshint ignore:end */
+//     })
+//     .catch(function (err) {
+//       return next(err);
+//     });
+// }
+
+/**** WORKS ***/
+
+function getWorks(req, res, next) {
+  var _id = (req.params.id);
+  req.app.locals.db.any(`
+    SELECT id, artist_id, title, width, height, depth, gallery, orientation, technique,
+           filename, filedata, year, price
+      FROM Works
+     WHERE artist_id = $1`, _id
+  )
+    .then(function (data) {
+      res.status(200)
+        .json({
+          works: data,
+        });
+    })
+    .catch(function (err) {
+      return next(err);
+    });
+}
+
+function getWork(req, res, next) {
+  var _id = (req.params.id);
+  req.app.locals.db.one(`
+    SELECT artist_id, title, width, height, depth, gallery, orientation, technique,
+           filename, filedata, year, price
+      FROM Works
+     WHERE id = $1`, _id
+  )
```

Comment:
Should require `raami_admin` rights, or check that the email address of `person_id` matches `req.session.user.email`.